fix: ensure migrate keeps inline/trailing comments in $props type definition and produces non-broken code

rgon opened the pull request:
Fixes #14140

Keeps inline comments when migrating types, as we already do with other statement types [L783](https://github.com/sveltejs/svelte/blob/551284ca2259d934d34b71a38071d85684bd6d4f/packages/svelte/src/compiler/migrate/index.js#L783C1-L783C43)
```ts
    export let step: number | null = null // In-line comments are now kept
```

```ts
    interface Props {
	step: number | null; // In-line comments are now kept
    }
```

Added tests for Typescript and JSDoc migration. In the latter, the comments are merged with leading comments. See `tests/migrate/jsdoc-with-comments/output.svelte`

dummdidumm commented:
Thank you! Could you add a changeset? Then we're good to merge

rgon commented:
Changeset added!

paoloricciuti commented:
Just one comment...while a bit weird unfortunately you can't assume trailing comments are only line comment. For example this

```svelte
<script>
	export let name = 'world'; /*
	* this is a trailing comment too
	**/
</script>
```
migrate to broken code
```svelte
<script>
	/**
	 * @typedef {Object} Props
	 * @property {string} [name] - * this is a trailing comment too
*
	 */

	/** @type {Props} */
	let { name = 'world' } = $props();
	* this is a trailing comment too
	**/
</script>
```
although to be fair it's broken in today migration too...but the comment should probably undergo the same "cleaning" process we apply to the leadingComments

rgon commented:
@paoloricciuti Nice catch! Absolutely fair to fix it while we're at it and add tests, especially when it produces broken code.

I have modified this PR with the fix (and reworked comment parsing a bit so that it's exactly like leadingComments) and added tests for the use case you mentioned. 

Adding a new patch fix to the changeset as well, since the issue predates my PR (and arguably is more important than the original issue!).

Rich-Harris commented:
preview: https://svelte-dev-git-preview-svelte-14143-svelte.vercel.app/

_this is an automated message_

paoloricciuti commented:
@rgon if you can fix linting we are good to merge

rgon commented:
The linting issues aren't in any of my touched files:

```

paoloricciuti commented:
> The linting issues aren't in any of my touched files:

It was a small issue in the migrate file...let's wait on all greens and i'll merge

paoloricciuti commented:
Thanks! 🚀
